testdata/testoutput2

```diff
@@ -19612,6 +19612,693 @@ Failed: error -49 at offset 3 in replacement: unknown substring
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    \x20
+ 0:  
+    ^
+ 0: ^
+    a
+ 0: a
+    b
+ 0: b
+\= Expect no match
+    c
+No match
+
+/[a-c--b]+/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[a-c]
+          cls:[b]
+          op: --
+        ]++
+        Ket
+        End
+------------------------------------------------------------------
+    ac
+ 0: ac
+    a
+ 0: a
+\= Expect no match
+    b
+No match
+
+/[a-c--b]{2,3}/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[a-c]
+          cls:[b]
+          op: --
+        ]{2,3}+
+        Ket
+        End
+------------------------------------------------------------------
+    ac
+ 0: ac
+    cac
+ 0: cac
+\= Expect no match
+    a
+No match
+    bb
+No match
+
+/x[a-c--b]+y/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        x
+        eclass[
+          cls:[a-c]
+          cls:[b]
+          op: --
+        ]++
+        y
+        Ket
+        End
+------------------------------------------------------------------
+    xacy
+No match
+    xaay
+No match
+    xay
+No match
+\= Expect no match
+    zacy
+No match
+    xacz
+No match
+    xy
+No match
+    xby
+No match
+
+/[a--/alt_extended_class
+Failed: error 106 at offset 4: missing terminating ] for character class
+
+/[a--a/alt_extended_class
+Failed: error 106 at offset 5: missing terminating ] for character class
+
+/[a--[a/alt_extended_class
+Failed: error 106 at offset 6: missing terminating ] for character class
+
+/[a--[a]/alt_extended_class
+Failed: error 212 at offset 7: missing terminating ] for character class (note '[' must be escaped under PCRE2_ALT_EXTENDED_CLASS)
+
+/[a--[a]--/alt_extended_class
+Failed: error 212 at offset 9: missing terminating ] for character class (note '[' must be escaped under PCRE2_ALT_EXTENDED_CLASS)
+
+/[a--]/alt_extended_class
+Failed: error 210 at offset 5: expected operand after operator in character class
+
+/[--a]/alt_extended_class
+Failed: error 209 at offset 3: unexpected operator in character class (no preceding operand)
+
+/[--]/alt_extended_class
+Failed: error 209 at offset 3: unexpected operator in character class (no preceding operand)
+
+/[a---b]/alt_extended_class
+Failed: error 208 at offset 5: invalid operator in character class
+
+/[a&&&b]/alt_extended_class
+Failed: error 208 at offset 5: invalid operator in character class
+
+/[a|||b]/alt_extended_class
+Failed: error 208 at offset 5: invalid operator in character class
+
+/[a||b&&c]/alt_extended_class
+Failed: error 211 at offset 7: brackets needed to clarify operator precedence in character class
+
+/[a--b&&c]/alt_extended_class
+Failed: error 211 at offset 7: brackets needed to clarify operator precedence in character class
```

Comment:
Why?

Reply:
Good question. I address this in the manpage updates included in the PR.

There's also the (overly?) detailed preliminary survey I did of other implementations of UTS#18 syntax, which I linked in an email thread: https://github.com/NWilson/pcre2/blob/user/niwilson/utr18/NEW-CLASSES.md

* UTS#18 doesn't define the relative precedence of the operators
* Some implementations ban mixing operators (ECMAScript spec requires this)
* Some implementations give them all the same precedence... but some implementations give them a different precedence

So it's a compatibility nightmare, what a mess. I don't think we can justify picking a "side" in the battle. It would be really unfortunate if we picked some precedence for the operators, and then they updated UTS#18 with a different choice.

Reply:
This choice definitely simplify the current code. However, perl supports multiple operators, so the code needs to be able to handle it. This might be another thing I need to implement.

Reply:
I'll be implementing the Perl syntax, don't worry. It will drop into the current parser very easily.